connectors/opa/opa_connector.go

```diff
@@ -58,266 +60,13 @@ func (s *server) GetPoliciesDecisions(ctx context.Context, in *pb.ApplicationCon
-}
-
-//Translate the evaluation received from OPA for (dataset, operation) into pb.OperationDecision
-func GetOPAOperationDecision(opaEval string, operation *pb.AccessOperation) (*pb.OperationDecision, error) {
-	resultInterface := make(map[string]interface{})
-	err := json.Unmarshal([]byte(opaEval), &resultInterface)
-	if err != nil {
-		log.Printf("error in unmarshaling opaEval into resultInterface: " + err.Error())
-		return nil, err
-	}
-	mainMap, ok := resultInterface["result"].(map[string]interface{})
-	if !ok {
-		log.Printf("error in format of OPA evaluation (incorrect result map)")
-		return nil, errors.New("error in format of OPA evaluation (incorrect result map)")
-	}
-
-	// log.Printf("************************** input: ", mainMap)
-
-	//Now iterate over
-	enforcementActions := make([]*pb.EnforcementAction, 0)
-	usedPolicies := make([]*pb.Policy, 0)
-
-	if mainMap["deny"] != nil {
-		lstDeny, ok := mainMap["deny"].([]interface{})
-		if !ok {
-			log.Printf("Error: unknown format of deny list")
-			return nil, errors.New("unknown format of deny content")
-		}
-		if len(lstDeny) > 0 {
-			newEnforcementAction := &pb.EnforcementAction{Name: "Deny", Id: "Deny-ID", Level: pb.EnforcementAction_DATASET, Args: map[string]string{}}
-			enforcementActions = append(enforcementActions, newEnforcementAction)
-
-			for i, reason := range lstDeny {
-				if reasonMap, ok := reason.(map[string]interface{}); ok {
-					if newUsedPolicy, ok := buildNewPolicy(reasonMap["used_policy"]); ok {
-						usedPolicies = append(usedPolicies, newUsedPolicy)
-						continue
-					}
-				}
-				log.Printf("Warning: unknown format of argument %d of lstDeny list. Skipping", i)
-				continue
-			}
-		}
-	}
-
-	if mainMap["transform"] != nil {
-		lstTransformations, ok := mainMap["transform"].([]interface{})
-		if !ok {
-			log.Printf("Error: unknown format of transformationss list")
-			return nil, errors.New("unknown format of transform content")
-		}
-		for i, transformAction := range lstTransformations {
-			newEnforcementAction, newUsedPolicy, ok := buildNewEnfrocementAction(transformAction)
-			if !ok {
-				log.Printf("Error: unknown format of transform action")
-				return nil, errors.New("unknown format of transform action")
-			}
-			enforcementActions = append(enforcementActions, newEnforcementAction)
-			if newUsedPolicy == nil {
-				log.Printf("Warning: empty used policy field for transformation %d", i)
-			} else {
-				usedPolicies = append(usedPolicies, newUsedPolicy)
-			}
-		}
-	}
 
-	if len(enforcementActions) == 0 { //allow action
-		newEnforcementAction := &pb.EnforcementAction{Name: "Allow", Id: "Allow-ID", Level: pb.EnforcementAction_DATASET, Args: map[string]string{}}
-		enforcementActions = append(enforcementActions, newEnforcementAction)
-	}
+	eval, err := s.opaReader.GetPoliciesDecisions(in, catalogConnectorAddress, timeOut)
+	jsonOutput, _ := json.MarshalIndent(eval, "", "\t")
```

Comment:
Shouldn't you check the err value first?

Reply:
fixed